lib/puppet/type/zpool.rb

```diff
@@ -90,6 +90,22 @@ def insync?(is)
       desc 'Log disks for this pool. This type does not currently support mirroring of log disks.'
     end
 
+    newproperty(:ashift) do
+      desc 'The Alignment Shift for the vdevs in the given pool.'
+
+      validate do |_value|
+        raise Puppet::Error _('This property is only supported on Linux and FreeBSD') unless Facter.value(:kernel) == 'Linux' || Facter.value(:kernel) == 'FreeBSD'
```

Comment:
Hum sorry, I just double-checked it live and on FreeBSD and the property is _not_ working, `ashift` has to be tweaked using a sysctl (which is beyond the scope of this module)…

```suggestion
        raise Puppet::Error _('This property is only supported on Linux') unless Facter.value(:kernel) == 'Linux'
```

Reply:
no problem :)